### PR TITLE
Add E2E tests for terms feature

### PR DESCRIPTION
**Pull request by burhandodhy**
### Description of the Change

This PR adds the E2E tests for the terms feature.

### Credits

Props @burhandodhy 

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my change.
- [x] All new and existing tests pass.

### CODE REVIEW

File: tests/cypress/integration/features/terms.spec.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+		});
+
+		cy.get('.ep-sync-progress strong', {
+			timeout: Cypress.config('elasticPressIndexTimeout'),
+		}).should('contain.text', 'Sync complete');
+
+		cy.wpCli('wp elasticpress list-features').its('stdout').should('contain', 'terms');
+	});
+
+	it('Can searching a term in the admin dashboard use Elasticsearch', () => {
```
Comment (felipeelia) [inline tests/cypress/integration/features/terms.spec.js]: @burhandodhy can we rephrase it to `Can search a term in the admin dashboard using Elasticsearch`?

File: tests/cypress/integration/features/terms.spec.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+		const term = 'amazing';
+
+		// Create a new term
+		cy.createTaxonomy({ name: term });
+
+		// Search for the term
+		cy.get('#tag-search-input').type(term);
+		cy.get('#search-submit').click();
+		cy.get('.wp-list-table tbody tr').should('have.length', 1).should('contain.text', term);
```
Comment (felipeelia) [inline tests/cypress/integration/features/terms.spec.js:76]: @burhandodhy can we check if the results came from Elasticsearch?

File: tests/cypress/integration/features/terms.spec.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+		// Delete the term
+		cy.get('.wp-list-table tbody tr')
+			.first()
+			.find('.row-actions .delete a')
+			.click({ force: true });
+
+		// Re-search for the term and make sure it's not there.
+		cy.get('#search-submit').click();
+		cy.get('.wp-list-table tbody').should('contain.text', 'No categories found');
```
Comment (felipeelia) [inline tests/cypress/integration/features/terms.spec.js:91]: @burhandodhy can we check if the results came from Elasticsearch?

File: tests/cypress/integration/features/terms.spec.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+		cy.get('#search-submit').click();
+
+		cy.get('.wp-list-table tbody tr')
+			.should('have.length', 1)
+			.should('contain.text', searchTerm);
+
+		cy.get('#debug-menu-target-EP_Debug_Bar_ElasticPress .ep-query-debug').should(
+			'contain.text',
+			'Query Response Code: HTTP 200',
+		);
```
Comment (felipeelia) [inline tests/cypress/integration/features/terms.spec.js]: @burhandodhy How could we make sure this query is really the one we are looking for?

File: tests/cypress/integration/features/terms.spec.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+		cy.wpCli('wp elasticpress list-features').its('stdout').should('contain', 'terms');
+	});
+
+	it('Can searching a term in the admin dashboard use Elasticsearch', () => {
+		cy.login();
+		cy.maybeEnableFeature('terms');
+
+		cy.visitAdminPage('edit-tags.php?taxonomy=category');
+
+		const searchTerm = 'classic';
```
Comment (felipeelia) [inline tests/cypress/integration/features/terms.spec.js]: @burhandodhy I think this comes from the default content being imported at the beginning of the tests, is that right? If so, can we create a term during the test? The initial import slows things down quite a bit and I'm planning to remove that eventually.

File: tests/cypress/integration/features/terms.spec.js [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+			cy.get('#tag-search-input').clear().type(tag);
+			cy.get('#search-submit')
+				.click()
+				.then(() => {
+					cy.get('.wp-list-table tbody tr')
+						.first()
+						.find('.row-actions .delete a')
+						.click({ force: true });
+				});
+		});
```
Comment (felipeelia) [inline tests/cypress/integration/features/terms.spec.js]: @burhandodhy although it sometimes can be counterintuitive, we should actually clean things created by tests *before* the tests: https://docs.cypress.io/guides/references/best-practices#Using-after-or-afterEach-hooks (the docs mention the hooks, but the idea remains the same.)

We could use something like we are using for users here: https://github.com/10up/ElasticPress/blob/develop/tests/cypress/integration/indexables/user.spec.js#L9

Thoughts?

File: tests/cypress/integration/features/terms.spec.js [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+		// create tags.
+		tags.forEach((tag) => {
+			cy.createTaxonomy({ name: tag, taxonomy: 'post_tag' });
+		});
+
+		// search for the tag.
+		cy.get('#tag-search-input').type('the most fun thing');
+		cy.get('#search-submit').click();
+
+		cy.get('.wp-list-table tbody tr .row-title').should('contain.text', 'The Most Fun Thing');
```
Comment (felipeelia) [inline tests/cypress/integration/features/terms.spec.js:113]: @burhandodhy can we check if the results came from Elasticsearch?

File: tests/cypress/integration/features/terms.spec.js [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+				.then(() => {
+					cy.get('.wp-list-table tbody tr')
+						.first()
+						.find('.row-actions .delete a')
+						.click({ force: true });
+				});
+		});
+	});
+
+	it('Can a child term be updated when a parent term is deleted', () => {
```
Comment (felipeelia) [inline tests/cypress/integration/features/terms.spec.js]: @burhandodhy can we rephrase this to something like `can update a child term when a parent term is deleted`? The intent is to come up with a phrase where `it` is the app doing something.

File: tests/cypress/support/commands.js [hunk 9]
```diff
@@ -57,8 +57,20 @@ Cypress.Commands.add('openWidgetsPage', () => {
 	});
 });
 
-Cypress.Commands.add('createTaxonomy', (name = 'Test taxonomy', taxonomy = 'category') => {
+Cypress.Commands.add('createTaxonomy', (data) => {
```
Comment (felipeelia) [inline tests/cypress/support/commands.js]: @burhandodhy great improvement here, thanks!

**Review by felipeelia — CHANGES_REQUESTED** [review]
Overall this looks great @burhandodhy! I left some comments and requested some changes but please let me know if you have any concerns or a different approach in mind.

File: tests/cypress/integration/features/terms.spec.js (hunk 6 above)
Comment (burhandodhy) [reply tests/cypress/integration/features/terms.spec.js]: @felipeelia as we are planning to get the rid of all the dummy data. Is it okay to delete all the terms by using the following command `wp term list post_tag --field=term_id | xargs wp term delete post_tag` in the before hook?

File: tests/cypress/integration/features/terms.spec.js (hunk 1 above)
Comment (burhandodhy) [reply tests/cypress/integration/features/terms.spec.js]: Done

File: tests/cypress/integration/features/terms.spec.js (hunk 2 above)
Comment (burhandodhy) [reply tests/cypress/integration/features/terms.spec.js:76]: Done

File: tests/cypress/integration/features/terms.spec.js (hunk 3 above)
Comment (burhandodhy) [reply tests/cypress/integration/features/terms.spec.js:91]: Done

File: tests/cypress/integration/features/terms.spec.js (hunk 4 above)
Comment (burhandodhy) [reply tests/cypress/integration/features/terms.spec.js]: Valid point. I changed the test and now we are looking for the search term in the Elasticsearch result.

File: tests/cypress/integration/features/terms.spec.js (hunk 5 above)
Comment (burhandodhy) [reply tests/cypress/integration/features/terms.spec.js]: Done. Created new term.

File: tests/cypress/integration/features/terms.spec.js (hunk 7 above)
Comment (burhandodhy) [reply tests/cypress/integration/features/terms.spec.js:113]: Done

File: tests/cypress/integration/features/terms.spec.js (hunk 8 above)
Comment (burhandodhy) [reply tests/cypress/integration/features/terms.spec.js]: Done

File: tests/cypress/integration/features/terms.spec.js (hunk 6 above)
Comment (burhandodhy) [reply tests/cypress/integration/features/terms.spec.js]: Thanks for suggestion. I moved it to after hook

File: tests/cypress/integration/features/terms.spec.js [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+		cy.get('.ep-feature-terms .settings-button').click();
+		cy.get('.ep-feature-terms [name="settings[active]"][value="1"]').click();
+		cy.get('.ep-feature-terms .button-primary').click();
+		cy.on('window:confirm', () => {
+			return true;
+		});
+
+		cy.get('.ep-sync-progress strong', {
+			timeout: Cypress.config('elasticPressIndexTimeout'),
+		}).should('contain.text', 'Sync complete');
```
Comment (felipeelia) [inline tests/cypress/integration/features/terms.spec.js]: @burhandodhy let's also check if the mapping was sent here, please. You can see how we are doing it for the WooCommerce feature [here](https://github.com/10up/ElasticPress/blob/develop/tests/cypress/integration/features/woocommerce.spec.js#L32)

File: tests/cypress/integration/features/terms.spec.js [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+			cy.get('#search-submit')
+				.click()
+				.then(() => {
+					cy.get('.wp-list-table tbody tr')
+						.first()
+						.find('.row-actions .delete a')
+						.click({ force: true });
+				});
+		});
+	});
```
Comment (felipeelia) [inline tests/cypress/integration/features/terms.spec.js]: @burhandodhy Can we move that to the before hook and use wp-cli to clean it up? With the change made on #2944 we should be able to call nested wp-cli commands now.

File: tests/cypress/support/commands.js (hunk 9 above)
Comment (felipeelia) [inline tests/cypress/support/commands.js]: @burhandodhy actually this was wrongly named before. Do you mind updating it to `createTerm`?

**Review by felipeelia — CHANGES_REQUESTED** [review]

File: tests/cypress/integration/features/terms.spec.js (hunk 6 above)
Comment (burhandodhy) [reply tests/cypress/integration/features/terms.spec.js]: Done.

File: tests/cypress/integration/features/terms.spec.js (hunk 10 above)
Comment (burhandodhy) [reply tests/cypress/integration/features/terms.spec.js]: Done

**Review by felipeelia — APPROVED** [review]
